podpac/core/interpolation/test/test_interpolation.py

```diff
@@ -333,3 +334,72 @@ def interpolate(self, udims, source_coordinates, source_data, eval_coordinates,
         outdata = interp.interpolate(srccoords, srcdata, reqcoords, outdata)
 
         assert np.all(outdata == srcdata)
+
+
+class TestHeterogenousInterpolation(object):
+    DATA = np.arange(64).reshape((4, 4, 4))
+    COORDS = Coordinates(
+        [[0, 1, 2, 3], [0, 1, 2, 3], ["2020-01-01", "2020-01-05", "2020-01-09", "2020-01-13"]],
+        dims=["lat", "lon", "time"],
+    )
+    C1 = Coordinates([1.0, 1.0, "2020-01-05"], dims=["lat", "lon", "time"])
+    C2 = Coordinates([1.2, 1.3, "2020-01-05"], dims=["lat", "lon", "time"])
+    C3 = Coordinates([1.0, 1.0, "2020-01-06"], dims=["lat", "lon", "time"])
+    C4 = Coordinates([1.2, 1.3, "2020-01-06"], dims=["lat", "lon", "time"])
+    C = Coordinates([[1.0, 1.2], [1.0, 1.3], ["2020-01-05", "2020-01-06"]], dims=["lat", "lon", "time"])
+
+    S1 = Coordinates([[1.0, 1.0], "2020-01-05"], dims=["lat_lon", "time"])
+    S2 = Coordinates([[1.2, 1.3], "2020-01-05"], dims=["lat_lon", "time"])
+    S3 = Coordinates([[1.0, 1.0], "2020-01-06"], dims=["lat_lon", "time"])
+    S4 = Coordinates([[1.2, 1.3], "2020-01-06"], dims=["lat_lon", "time"])
+    S = Coordinates(
+        [[[1.0, 1.2, 1.0, 1.2], [1.0, 1.3, 1.0, 1.3]], ["2020-01-05", "2020-01-06"]], dims=["lat_lon", "time"]
+    )
+
+    def test_nearest(self):
+        interpolation = "nearest"
+        node = podpac.data.Array(source=self.DATA, coordinates=self.COORDS, interpolation=interpolation)
+
+        assert node.eval(self.C1)[0, 0, 0] == 21.0
+        assert node.eval(self.C2)[0, 0, 0] == 21.0
+        assert node.eval(self.C3)[0, 0, 0] == 21.0
+        assert node.eval(self.C4)[0, 0, 0] == 21.0
+        assert np.all(node.eval(self.C) == 21.0)
+
+        assert node.eval(self.S1)[0, 0] == 21.0
+        assert node.eval(self.S2)[0, 0] == 21.0
+        assert node.eval(self.S3)[0, 0] == 21.0
+        assert node.eval(self.S4)[0, 0] == 21.0
+        np.testing.assert_array_equal(node.eval(self.S), [[21, 21], [21, 21], [21, 21], [21, 21]])
+
+    def test_mixed(self):
+        interpolation = [{"method": "nearest", "dims": ["time"]}, {"method": "bilinear", "dims": ["lat", "lon"]}]
```

Comment:
Could you also add a case with 'bilinear' for time and "nearest" for the other two? Does that work or not implemented?

I'm curious if it works "bilinear" for lat and "nearest" for time. If it doesn't that's fine, we can skip that. I'm just curious if that would work.

Reply:
> Could you also add a case with 'bilinear' for time and "nearest" for the other two? Does that work or not implemented?

As far as I can tell, podpac doesn't do linear interpolation for time at all.

`interpolation='bilinear'` gives an exception:

```
>>> node = podpac.data.Array( 
...     source=[0, 1, 2], 
...     coordinates=podpac.Coordinates([["2020-01-01", "2020-01-05", "2020-01-09"]], dims=['time']), 
...     interpolation="bilinear" 
... ) 
>>> node.eval(podpac.Coordinates([podpac.crange("2020-01-01", "2020-01-09", "1,D")], dims=['time']))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jmilloy/Creare/Pipeline/podpac/podpac/core/node.py", line 984, in wrapper
    data = fn(self, coordinates, output=output)
  File "/home/jmilloy/Creare/Pipeline/podpac/podpac/core/data/datasource.py", line 480, in eval
    output = self._interpolation.interpolate(
  File "/home/jmilloy/Creare/Pipeline/podpac/podpac/core/interpolation/interpolation.py", line 490, in interpolate
    interpolator_queue = self._select_interpolator_queue(
  File "/home/jmilloy/Creare/Pipeline/podpac/podpac/core/interpolation/interpolation.py", line 393, in _select_interpolator_queue
    raise InterpolationException(
podpac.core.interpolation.interpolation.InterpolationException: Dimensions ['time'] can't be handled by interpolation definition:
 Interpolation
	('default',): bilinear, ['Rasterio', 'ScipyGrid'], {}
```

And `interpolation=[{"method": "bilinear", "dims": ["time"]}]` is silently ignored in favor of nearest.

```
>>> node = podpac.data.Array(
...     source=[0, 1, 2], 
...     coordinates=podpac.Coordinates([["2020-01-01", "2020-01-05", "2020-01-09"]], dims=['time']), 
...     interpolation=[{"method": "bilinear", "dims": ["time"]}]
... )
>>> node.eval(podpac.Coordinates([podpac.crange("2020-01-01", "2020-01-09", "1,D")], dims=['time']))
<xarray.UnitsDataArray (time: 9)>
array([0., 0., 1., 1., 1., 1., 2., 2., 2.])
Coordinates:
  * time     (time) datetime64[ns] 2020-01-01 2020-01-02 ... 2020-01-09
Attributes:
    layer_style:  <podpac.core.style.Style object at 0x7f775e6c0c40>
    crs:          EPSG:4326
```
node = podpac.data.Array(
            source=[0, 1, 2],
            coordinates=Coordinates([["2020-01-01", "2020-01-05", "2020-01-09"]], dims=['time']),
            interpolation="bilinear"
        )
        np.testing.assert_array_equal(
            node.eval(Coordinates([podpac.crange("2020-01-01", "2020-01-09", "1,D")], dims=['time'])),
            [0.00, 0.25, 0.50, 0.75, 1.00, 1.25, 1.50, 1.75, 2.00]
        )
```

I'll make an issue for that.

> I'm curious if it works "bilinear" for lat and "nearest" for time. If it doesn't that's fine, we can skip that. I'm just curious if that would work.

Is this a typo? Or can you clarify?

Reply:
You'll have seen by now that I started to make an issue, and then made several, but they still don't cover everything. I don't actually think it will be too difficult to make the existing interpolation cover more cases, nor to add the xarray interpolator which will be more general. In any case, I added this case.

Reply:
Yep that was a typo, should have been "bilinear" for lat and "nearest" for **lon**

Reply:
Ah, so I think this is clear from comments above, but that won't work for the same reasons -- it'll silently use nearest lat even though you specified bilinear.

Reply:
Good to know. Thanks.